text/0510-dynamodb-global-table.md

```diff
@@ -0,0 +1,1742 @@
+
+### DynamoDB Global TableKMS
+
+[DynamoDB Global Table](https://aws.amazon.com/dynamodb/global-tables/) lets you provision a table that could be
+ replicated across different regions. It could also be deployed to just one region and would cost the same as a
+ single DynamoDB table.
+
+#### Partition Key
+
+Our GlobalTable L2 construct could be used to provision a global table. You can do this as follows,
+
+```typescript
+  new GlobalTable(stack, 'FooTable', {
+    partitionKey: { name: 'Foo', type: AttributeType.STRING },
+  });
+```
+
+This would provision a global table only in the region where the stack is being deployed to. The provisioning mode
+ would be on-demand. And, it would also add a partition key which would be the same for all replicas of the table.
+
+#### Sort Key
+
+You can add a sort key to your global table. This sort key would be the same across all replicas of the table.
+
+```typescript
+  new GlobalTable(stack, 'FooTable', {
+    partitionKey: { name: 'Foo', type: AttributeType.STRING },
+    sortKey: { name: 'Bar', type: AttributeType.STRING },
+  });
+```
+
+#### Table Name
+
+You can mention a name for your global table. If not specified, this is auto generated by cloudformation. This
+would be the same for each replica.
+
+```typescript
+  new GlobalTable(stack, 'FooTable', {
+    partitionKey: { name: 'Foo', type: AttributeType.STRING },
+    tableName: 'FooBarTable',
+  });
+```
+
+#### Billing Mode
+
+There are two billing modes that global tables could have,
+
+1. Pay Per Request: In this mode you would not need to specify write or read capacity. You would only be
+paying for the capacity utilized by the requests coming in.
+2. Provisioned: In this mode you would have write with auto scaling and read capacity with either auto
+scaling or with exact read capacity units. You would be paying for the provisioned capacity even if it is
+not utilized.
+
+The default mode is pay per request mode. If you choose provisioned mode, then you would need to specify read
+ and write capacity for the table and replicas. You can do this in three ways,
+
+1. Mention capacity on table level and these get copied over to all replicas.
+2. Mention capacity for each replica separately.
+3. Mention capacity on table level and for replica where you want to have a different capacity. Wherever you
+would not mention capacity, the table level values would be used.
+
+The billing mode will remain the same across all replicas.
+
+##### Pay Per Request Mode
+
+```typescript
+  new GlobalTable(stack, 'FooTable', {
+    partitionKey: { name: 'Foo', type: AttributeType.STRING },
+    billingMode: BillingMode.PAY_PER_REQUEST,
+  });
+```
+
+##### Provisioned Mode
+
+* __Table level configuration with auto scaling__
+
+```typescript
+  new GlobalTable(stack, 'FooTable', {
+    partitionKey: { name: 'Foo', type: AttributeType.STRING },
+    billingMode: BillingMode.PROVISIONED,
+    writeWithAutoScaling: {
```

Comment:
`withAutoScaling`? 

There is no non-autoscaling read/write, so seems like you don't have to provide this?